src/briefcase/integrations/subprocess.py

```diff
@@ -206,6 +210,39 @@ def Popen(self, args, **kwargs):
             **self.final_kwargs(**kwargs)
         )
 
+    def stream_output(self, popen_process):
+        """
+        Stream the output of a Popen process until the process exits.
+        If the user sends CTRL+C, the process will be terminated.
+
+        This is useful for starting a process via Popen such as tailing a
+        log file, then initiating a non-blocking process that populates that
+        log, and finally streaming the original process's output here.
+
+        :param popen_process: a running Popen process with output to print
+        """
+        try:
+            while True:
+                # readline should always return at least a newline...if it returns an
+                # empty string, that should mean the underlying process is exiting/gone
+                output_line = ensure_str(popen_process.stdout.readline())
+                return_code = popen_process.poll()
+                if output_line == "" and return_code is not None:
+                    self._log_return_code(return_code)
+                    break
+                if output_line != "":
+                    self.command.logger.info(output_line)
+
+        except KeyboardInterrupt:
+            popen_process.terminate()
+            try:
+                popen_process.wait(timeout=3)
+            except subprocess.TimeoutExpired:
+                popen_process.kill()
+
+        except Exception as e:
+            raise PopenStreamingError(f"{e!r}")
```

Comment:
On one hand, this sort of catching feels somewhat overdone seems Im really not sure how readline or poll can error....on the other hand, re-wrapping whatever might come out in a more contextual error seems appropriate.

Reply:
That's a good perspective. Removed the catch.